rfc/0084-app-service-level-telemetry.md

```diff
@@ -0,0 +1,491 @@
+other types of errors.
+
+For this set of metrics, we propose gathering simple counters, and on a per SDK
+instance basis (these do not correlate 1:1 to hostnames and a definition of
+runtime/instance identifiers is in the [Response Time Observability
+RFC](rfc/0035-rto.md)). Each counter must be associated with the connection and
+bucket if it is possible. In particular, it must be possible to label counters
+with bucket name and hostname for KV service, while often not possible to even
+reliably associate hostname with Search request due to proxies on the Cloud
+deployments.
+
+From these two simple pieces of information, we can intuit that the connection
+was likely severed and a number of operations timed out before the connection
+could be re-established.
+
+Determining why the connection was severed requires correlation to other
+information.  For example, we may need to look at the Couchbase Server logs to
+see that a particular process for a given service crashed where we would expect
+to find the SDK runtime/instance identifier being sent in the request (defined
+in the [Response Time Observability RFC](rfc/0035-rto.md)).
+
+We should note that existence of timeouts is possibly expected.  For example,
+with an ad-targeting or recommendation targeting deployment, there may be short
+latencies that when not serviced, lead to other processing.  This is worth
+mentioning because it will be important to compare a deployment to itself, and
+there may be limited value in comparing across deployments.
+
+## Capella Service Level Metrics
+
+The counters described above give a good picture of actual impact on a deployed
+app, but they do not give a sense of whether or not a given deployment is
+meeting an expected service level.  Said another way, a Couchbase Capella
+Cluster deployment could be operational, but very slow during a rebalance
+leading to user discontent, but never turn into an error.
+
+Service levels are specific to the user's targets.  While it's impossible to
+properly measure whether or not an end-user's app is meeting service level
+targets, we can establish a predefined histogram that would allow Couchbase to
+simply and inexpensively understand if the service level has likely changed.
+
+It's proposed that we establish a set of standardized histograms, specific to
+services and to the service level expected with the service using those
+parameters.  Counters for these histograms can be sent to Couchbase and both
+stored individually and aggregated. See histogram buckets in "Histograms" table
+below.
+
+As an aggregated set of metrics, it would give Couchbase a rough view over the
+service level the user is experiencing.  Deviations in service level can be dug
+into and, in particular, we can check for correlation to Couchbase Capella
+management activities like upgrades.
+
+The Indexing Service and Eventing Services are not included as they are not
+network-available services for applications to make requests from.
```

Comment:
The Eventing service is network-available. We have a number of eventing management operations. If we're recording (for example) search management operations, is there a reason we shouldn't do the same for eventing?

Reply:
Good point. I will update this part to say that we should only exclude services that we do not touch. Like indexing service